test_run.lua

```diff
@@ -284,6 +285,26 @@ local function grep_log(self, node, what, bytes, opts)
     return found
 end
 
+-- Checks fn with delay until timeout or fn returns true, otherwise returns false
```

Comment:
Nit: Our code style suggests to keep comments within 66 symbols, start it with an uppercase letter and end with a period.